api/controllers/StatsController.js

```diff
@@ -96,20 +102,76 @@ function transformSeasonToDTO(season) {
   return {
     ...rest,
     rankings: rankingsAsArray,
+    uniquePlayersPerWeek: season.uniquePlayersPerWeek.map((playerSet) => playerSet.size),
   };
 }
 
+/**
+ * Returns copy of seasons with the initialized arrays for
+ * gameCounts and uniquePlayersPerWeek
+ * @param {
+*  name: string
+*  startTime: int
+*  endTime: int
+*  firstPlace: int (playerId)
+*  secondPlace: int (playerId)
+*  thirdPlace: int (playerId)
+*  fourthPlace: int (playerId)
+*  rankings: Map<int: playerId, {
+*    username: string
+*    matches: Map<int: weekNum, Array<{opponent: string, result: Result}>>
+*  }>
+* } seasons
+* 
+* @returns {
+*  name: string
+*  startTime: int
+*  endTime: int
+*  firstPlace: int (playerId)
+*  secondPlace: int (playerId)
+*  thirdPlace: int (playerId)
+*  fourthPlace: int (playerId)
+*  rankings: Map<int: playerId, {
+*    username: string
+*    matches: Map<int: weekNum, Array<{opponent: string, result: Result}>>
+*  }>
+*  gameCounts: Array<int>
+*  uniquePlayersPerWeek: Array<Set<int: playerIds>>
+* }
+*/
+function initializePlayerAndGameCounts(seasons) {
+  return seasons.map((season) => {
+    const res = {...season};
+    res.gameCounts = [];
+    res.uniquePlayersPerWeek = [];
+    const startTime = dayjs(season.startTime);
+    const endTime = dayjs(season.endTime);
+    const numWeeks = endTime.diff(startTime, 'week');
+    for (let i=0; i<numWeeks; i++) {
+      res.gameCounts.push(0);
+      res.uniquePlayersPerWeek.push(new Set());
+    }
+    return res;
+  });
+}
```

Comment:
Done. I pulled the logic for initializing the `uniquePlayersPerWeek` and `gameCounts` arrays into the `getSeasonsWithoutRankings` helper. This help keeps the type signatures more consistent so we aren't adding properties at different stages (now there's the one type used during processing and the transformed version returned by the API), plus we skip an extra loop by consolidating the initialization in the one `getSeasonsWithoutRankings()` helper rather than looping there and also in the controller